src/SmartComponents/DriftPage/DriftTable/DriftTable.js

```diff
@@ -177,17 +210,28 @@ export class DriftTable extends Component {
     findSystem(fact) {
         let row = [];
         let system = undefined;
+        let className;
 
         this.masterList.forEach(item => {
             system = fact.systems.find(function(sys) {
                 return sys.id === item.id;
             });
 
-            row.push(
-                <td className={ fact.state === 'DIFFERENT' ? 'highlight comparison-cell' : 'comparison-cell' }>
-                    { system.value === null ? 'No Data' : system.value }
-                </td>
-            );
+            if (this.referenceId) {
+                if (system.state === 'DIFFERENT') {
+                    className = 'highlight comparison-cell different-fact-cell';
+                } else {
+                    className = 'comparison-cell';
+                }
+            } else {
+                if (fact.state === 'DIFFERENT') {
+                    className = 'highlight comparison-cell';
+                } else {
+                    className = 'comparison-cell';
+                }
+            }
```

Comment:
Note that the code adds `comparison-cell` in all cases. The `else` clauses could be dropped by starting with that:

```
let className = ['comparison-cell']

if (fact.state === 'DIFFERENT') {
     className.push('highlight');

     if (this.referenceId) {
         className.push('different-fact-cell');
}

row.push(<td className={ className,join() }>{ system.value === null ? 'No Data' : system.value }</td>);
```

Reply:
Great change!